lib/Twig/Extensions/TokenParser/Trans.php

```diff
@@ -37,13 +38,42 @@ public function parse(Twig_Token $token)
                 $stream->expect(Twig_Token::BLOCK_END_TYPE);
                 $plural = $this->parser->subparse(array($this, 'decideForFork'));
 
-                if ('notes' === $stream->next()->getValue()) {
+                $next_after_plural = $stream->next()->getValue();
+                if ('notes' === $next_after_plural) {
                     $stream->expect(Twig_Token::BLOCK_END_TYPE);
-                    $notes = $this->parser->subparse(array($this, 'decideForEnd'), true);
+                    $notes = $this->parser->subparse(array($this, 'decideForFork'));
+
+                    if('context' === $stream->next()->getValue()){
+                        $stream->expect(Twig_Token::BLOCK_END_TYPE);
+                        $context = $this->parser->subparse(array($this, 'decideForEnd'), true);
+                    }
+                }
+                elseif('context' === $next_after_plural){
+                    $stream->expect(Twig_Token::BLOCK_END_TYPE);
+                    $context = $this->parser->subparse(array($this, 'decideForFork'));
+
+                    if ('notes' === $stream->next()->getValue()) {
+                        $stream->expect(Twig_Token::BLOCK_END_TYPE);
+                        $notes = $this->parser->subparse(array($this, 'decideForEnd'), true);
+                    }
                 }
             } elseif ('notes' === $next) {
                 $stream->expect(Twig_Token::BLOCK_END_TYPE);
-                $notes = $this->parser->subparse(array($this, 'decideForEnd'), true);
+                $notes = $this->parser->subparse(array($this, 'decideForFork'));
+
+                if('context' === $stream->next()->getValue()){
+                    $stream->expect(Twig_Token::BLOCK_END_TYPE);
+                    $context = $this->parser->subparse(array($this, 'decideForEnd'), true);
+                }
+            }
+            elseif('context' === $next){
+                $stream->expect(Twig_Token::BLOCK_END_TYPE);
+                $context = $this->parser->subparse(array($this, 'decideForFork'));
+
+                if ('notes' === $stream->next()->getValue()) {
+                    $stream->expect(Twig_Token::BLOCK_END_TYPE);
+                    $notes = $this->parser->subparse(array($this, 'decideForEnd'), true);
+                }
```

Comment:
This is done in order to be able to use such calls:
1) trans -> notes -> context
2) trans -> context -> notes
3) trans -> plural -> context -> notes
4) trans -> plural -> notes -> context